### ISSUE TITLE
Using a remote tilejson spec behaves differently than inline: ignores scheme

### ISSUE THREAD

**Issue by gampleman** [mapbox/mapbox-gl-js#6526]
Compare:
1. https://jsbin.com/pokilokeca/edit?html,output
2. https://jsbin.com/mosuxizata/edit?html,output

The first uses a remote tileJSON spec, the second inlines the same tileJSON spec into the `source` declaration. 

I would expect these to behave identically, but the first seems to ignore the `scheme: 'tms'` parameter when it comes from the remote definition.

**Comment by jfirebaugh** [comment]
Indeed, #2565 added support for `scheme` in the wrong place; it should be handled in `loadTileJSON`.
